app/src/androidTest/java/at/tugraz/ist/swe/cheatapp/DatabaseIntegrationTest.java

```diff
@@ -0,0 +1,118 @@
+package at.tugraz.ist.swe.cheatapp;
+
+import android.arch.lifecycle.Observer;
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class DatabaseIntegrationTest {
+    private MessageRepository messageRepository;
+    private Context context;
+
+    private void deleteDatabase(Context appContext) {
+        File databases = new File(appContext.getApplicationInfo().dataDir + "/databases");
+        File db = new File(databases, "cheatapp_db");
+        if (db.delete())
+            System.out.println("Database deleted");
+        else
+            System.out.println("Failed to delete database");
+    }
+
+    @Before
+    public void setUp() {
+        context = InstrumentationRegistry.getTargetContext().getApplicationContext();
+        deleteDatabase(context);
+        messageRepository = new MessageRepository(context);
+    }
+
+    @Test
+    public void testInitialization() {
+        assertNotNull(messageRepository);
+    }
+
+    @Test
+    public void testInsertAndLoadOneMessage() throws InterruptedException {
+        messageRepository.insertMessage(0, "First Message!", true);
+
+        Thread.sleep(100);
+
+        List<Message> allMessages = messageRepository.getRawMessagesByUserId(0);
+        Message message = allMessages.get(0);
+
+        assertEquals(1, allMessages.size());
+        assertEquals("First Message!", message.getMessageText());
+        assertEquals(0, message.getUserId());
+        assertEquals(true, message.getMessageSent());
+        assertEquals(1, message.getMessageId());
+    }
+
+    @Test
+    public void testInsertAndLoadMultipleMessages() throws InterruptedException {
+        messageRepository.insertMessage(1, "First Message!", true);
```

Comment:
Personally I would pass a Message instance to this method instead of passing the raw parameters